### PR TITLE
Update blittable concept for clarity with UnmanagedCallersOnlyAttribute.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
## Summary

Clarify the definition of a "blittable" type and split the "contains blittable contents" concept from the "blittable type" concept to improve clarity around the requirements of argument and return types for UnmanagedCallersOnlyAttribute-marked methods.

### CODE REVIEW

File: docs/standard/native-interop/best-practices.md [hunk 1]
```diff
@@ -82,25 +82,35 @@ GUIDs are usable directly in signatures. Many Windows APIs take `GUID&` type ali
...
 
-- `char`, `string`
+- `char`
+
+**Types with SOMETIMES blittable contents:**
+
+- `string`
 
-When blittable types are passed by reference, they're simply pinned by the marshaller instead of being copied to an intermediate buffer. (Classes are inherently passed by reference, structs are passed by reference when used with `ref` or `out`.)
+When blittable types are passed by reference with `in`, `ref`, or `out`, or when types with blittable contents are passed by value, they're simply pinned by the marshaller instead of being copied to an intermediate buffer.
```
Comment (AaronRobinsonMSFT) [inline docs/standard/native-interop/best-practices.md:113]: This is only true when the desired `CharSet` matches the runtime encoding (i.e. `CharSet.Unicode`). Unless I am missing what is being stated here.

Comment (AaronRobinsonMSFT) [reply docs/standard/native-interop/best-practices.md:113]: I see you mentioned this below. Perhaps clarify here as well.

File: docs/standard/native-interop/best-practices.md [hunk 2]
```diff
@@ -112,9 +122,9 @@ public struct UnicodeCharStruct
 }
 ```
 
-`string` is blittable if it isn't contained in another type and it's being passed as an argument that is marked with `[MarshalAs(UnmanagedType.LPWStr)]` or the `[DllImport]` has `CharSet = CharSet.Unicode` set.
+`string` contains blittable contents if it isn't contained in another type and it's being passed as an argument that is marked with `[MarshalAs(UnmanagedType.LPWStr)]` or the `[DllImport]` has `CharSet = CharSet.Unicode` set.
 
-You can see if a type is blittable by attempting to create a pinned `GCHandle`. If the type isn't a string or considered blittable, `GCHandle.Alloc` will throw an `ArgumentException`.
+You can see if a type is blittable or contains blittable contents by attempting to create a pinned `GCHandle`. If the type isn't a string or considered blittable, `GCHandle.Alloc` will throw an `ArgumentException`.
```
Comment (AaronRobinsonMSFT) [inline docs/standard/native-interop/best-practices.md:127]: I personally prefer the `Marshal.SizeOf()` method as it has less side effects. Unless that doesn't handle all the known cases.

Comment (jkoritzinsky) [reply docs/standard/native-interop/best-practices.md:127]: Marshal.SizeOf works for non-blittable value types as well so we can’t use it.

**Comment by gewarren** [comment]
@jkoritzinsky Is this ready to go?

**Comment by jkoritzinsky** [comment]
Yes it's ready to go.

**Review by gewarren — APPROVED** [review]

**Pull request merged**